src/can/can.rs

```diff
@@ -0,0 +1,161 @@
+use super::enums::*;
+use super::CanFrame;
+use crate as hal;
+use crate::can::registers::Registers;
+use crate::can::util;
+use crate::{into_ref, pac, Peripheral, PeripheralRef, RccPeripheral};
+
+pub struct Can<'d, T: Instance> {
+    _peri: PeripheralRef<'d, T>,
+    fifo: CanFifo,
+    last_mailbox_used: usize,
+}
+
+#[derive(Debug)]
+pub enum CanInitError {
+    InvalidTimings,
+}
+
+impl<'d, T: Instance> Can<'d, T> {
+    /// Assumes AFIO & PORTB clocks have been enabled by HAL.
+    ///
+    /// CAN_RX is mapped to PB8, and CAN_TX is mapped to PB9.
+    pub fn new(
+        peri: impl Peripheral<P = T> + 'd,
+        rx: impl Peripheral<P = impl RxPin<T>> + 'd,
+        tx: impl Peripheral<P = impl TxPin<T>> + 'd,
+        fifo: CanFifo,
+        mode: CanMode,
+        bitrate: u32,
+    ) -> Result<Self, CanInitError> {
+        into_ref!(peri, rx, tx);
+
+        let this = Self {
+            _peri: peri,
+            fifo,
+            last_mailbox_used: usize::MAX,
+        };
+        T::enable_and_reset(); // Enable CAN peripheral
+
+        rx.set_mode_cnf(
+            pac::gpio::vals::Mode::INPUT,
+            pac::gpio::vals::Cnf::PULL_IN__AF_PUSH_PULL_OUT,
```

Comment:
Leave these pin configs to me. I'll get it implemented.

Reply:
I don't mind. What should I do?)